lib/tokens/CookieTokenManager.js

```diff
@@ -26,28 +26,48 @@ class CookieTokenManager extends TokenManager {
 
   _getToken(callback) {
     debug('Submitting cookie token request.');
+    let boundedCompleted = this._reqCompleted.bind({
+      cb: callback
+    });
+
+    let protocol;
+    if (this._sessionUrl.startsWith('https')) {
+      protocol = require('https');
+    } else {
```

Comment:
Not really. There are protocol checks at these parts of the code:
 - https://github.com/cloudant/nodejs-cloudant/blob/35dbb085857bb423bc86aef10c4e27a302e1f113/cloudant.js#L63-L65
 - https://github.com/cloudant/nodejs-cloudant/blob/35dbb085857bb423bc86aef10c4e27a302e1f113/lib/client.js#L188-L193

I tried to use the same logic. But I can replace the url check with this solution (just to be consistent):
https://github.com/cloudant/nodejs-cloudant/blob/35dbb085857bb423bc86aef10c4e27a302e1f113/cloudant.js#L63

Reply:
Since a possible exception is not handled at the entry point of the client I guess it is not your task to handle it here. I am ok with it if you leave this as-is.

Reply:
I changed in 89d3fb0eb29f3b36000ddc34844c3eb8757cf640